### ISSUE TITLE
Support for Nokia 

### ISSUE THREAD

**Issue by laimaretto** [Vadims06/topolograph#5]
Hi there, I tried following the link to generate the textFSM for new vendors, but it's broken.

Still, do you have any plans in supporting nokia? How can I help?

thanks!

**Comment by Vadims06** [comment]
Hi @laimaretto. Thank you for your request! As I commented on YouTube page - TextFSM templates have been prepared by one of topolograph's user, and we would like to test them on some of test nokia's LSDB output. Could you please drop me test LSDB output to admin@topolograph.com and I would happy to add support Nokia to Topolograph =)

**Comment by laimaretto** [comment]
Hey! Thanks for the fast response ... let me get the output of an LSDB from Nokia and I'll share that with you ...

The broken link is in the main page of the project. An image is worth a thousand words ... :-)

![image](https://user-images.githubusercontent.com/8891825/105501878-87a38f80-5ca3-11eb-8542-be5e6c6aba3b.png)

**Comment by Vadims06** [comment]
Hi @laimaretto. Wiki link has been repaired. As well as @hellt pushed his textfsm templates for Nokia to textfsm_templates folder. We are ready to test them on a sample of LSDB. Commands are:

- show router ospf database type router detail
- show router ospf database type network detail
- show router ospf database type external detail

**Comment by laimaretto** [comment]
Hi @Vadims06 , thanks for the reply. Sorry for the delay. I've sent to the provided email admin@topolograph.com the outputs for the suggested `show` commands and a plot of the topology for your reference. I have added a also a show for `summary` type LSAs (LSA3). All the output gathered at router `SR2`. The topology of testing is the following:

![image](https://user-images.githubusercontent.com/8891825/105997507-73d4a080-608a-11eb-9024-72d20b0c58f9.png)

So, in the email you will find 4 files, one per each of your suggestion plus the one for summary LSAs.

Let me know if this works or even if you would like to make any modifications to the topology.

**Comment by Vadims06** [comment]
@laimaretto Thank you for your output and the topology. Could I ask you to make sure that the link between SR3-ABR and SR4-ASBR has OSPF type broadcast. And then send the command outputs from SR3-ABR. Thank you again.

**Comment by laimaretto** [comment]
hi @Vadims06 !. So, I've made the suggested modification. I didn't use before the broadcast type, since we're used to work with point-to-point interfaces. So I've set up broadcast between SR3-ABR and SR4-ASBR. I've added the IP addresses to the topology for your reference. And I'm sending right away the logs to the email, obtained at SR3-ABR. Thanks!

![image](https://user-images.githubusercontent.com/8891825/106148638-0b53f500-6158-11eb-95e3-c9bc1b21e772.png)

**Comment by laimaretto** [comment]
Hi @Vadims06 , just one further question. As you mentioned before, the templates for nokia are already in the code (it can be seen that "Vadims06 Merge pull request #6 from hellt/nokia-sros 002f6af). However, if I want to try these DBs outputs, on the website of topolograph (https://topolograph.com/upload-ospf-lsdb) I have no option for Nokia.

![image](https://user-images.githubusercontent.com/8891825/106324001-2194aa80-6257-11eb-99f0-115c30958ef8.png)

Finally, if I download the docker version (not yet, I'm learning how to docker), will this be available there as well? Thanks!

**Comment by Vadims06** [comment]
Good morning @laimaretto , yes, Nokia's templates have been uploaded and I'm going to add them on the site on this weekends. They have to be added manually)

**Comment by Vadims06** [comment]
@laimaretto looks like it works!)
![image](https://user-images.githubusercontent.com/20796986/106398812-0af07e00-6426-11eb-802c-4972ca07b19c.png)
Additionally, I pushed a container topolograph:2.14 with Nokia support, but I didn't make it as latest. So you can try upload a topology online or specify topolograph:2.14 in docker-compose.yml.
I'm looking forward to hearing updates from you,

**Comment by laimaretto** [comment]
hi @Vadims06 , very nice, thank you very much! I have indeed tried out the outputs from the test topology (on the web site), and I get the plot and I can do some analysis on the network. Beautiful. I still need to understand how to make the docker application work in my PC, I want to try that out as well. And I will let you know. For the time being, I'll be using the web application. Thanks again! Cheers! Lucas

**Comment by hellt** [comment]
@Vadims06 I think we need to add instructions on how to created OSPF LSDB for various vendors.
I.e. how do I know what show commands output should I put in a text file to upload for graph visualization for Nokia for example?

**Comment by Vadims06** [comment]
@hellt, I always open-minded for any improvements)

> how to create OSPF LSDB.

Good point, I tried to keep the table with all commands updated. The table exists on the main page of this repo
![image](https://user-images.githubusercontent.com/20796986/106465280-d1158b00-64aa-11eb-8ea8-edc860694965.png) or on the [site](https://topolograph.com/how-to#GeneralView-0). Probably it's better to rename these tables or create a separate wiki page for this. Any thoughts?..
I will add nokia's commands right now.

**Comment by laimaretto** [comment]
Hi @Vadims06 , just to tell you that I've installed the docker application, keeping in mind to  change `latest` to `2.14` and it works really nice. Beautil tool ... thanks for this! :-) ... I'll be using it with bigger databases and let you know in any case or if I see something ... thanks again!

**Comment by hellt** [comment]
@laimaretto @Vadims06 I wonder if we need to add multiple show commands output in the same file (per LSA type) before uploading it to the site?

**Comment by Vadims06** [comment]
@laimaretto glad to know! The next step is to implement napalm support for Nokia and i will try to add this into the container later.

**Comment by Vadims06** [comment]
> @laimaretto @Vadims06 I wonder if we need to add multiple show commands output in the same file (per LSA type) before uploading it to the site?

@hellt exactly. ‘show... type router, network and external‘ should be placed in a single file.

**Comment by hellt** [comment]
got it @Vadims06 

will spread the word inside the company to see how ppl will use it and if they will discover any issues

**Comment by rp-nz** [comment]
> @laimaretto looks like it works!)
> ![image](https://user-images.githubusercontent.com/20796986/106398812-0af07e00-6426-11eb-802c-4972ca07b19c.png)
> Additionally, I pushed a container topolograph:2.14 with Nokia support, but I didn't make it as latest. So you can try upload a topology online or specify topolograph:2.14 in docker-compose.yml.
> I'm looking forward to hearing updates from you,

Hi there where abouts in the docker-compose.yml do i specify the latest version to be 2.14. Can you please provide the syntax. I am getting errors when I try to pull the down the image due to docker syntax issue. 

Thanks

**Comment by Vadims06** [comment]
Hi @rp-nz,
the syntax should be like this 
```

  flask:
    container_name: flask
    image: vadims06/topolograph:2.14
    restart: unless-stopped
    environment:```

**Comment by rp-nz** [comment]
> Hi @rp-nz,
> the syntax should be like this
> 
> ```
> 
>   flask:
>     container_name: flask
>     image: vadims06/topolograph:2.14
>     restart: unless-stopped
>     environment:```
> ```

Thanks that did the trick. Trying out SROS OSPF DB now. Seems to be working pretty good.

**Comment by rp-nz** [comment]
Are there plans to support IS-IS DB to this tool and also any way we can add node names to each of the nodes in the DB.

Thanks

**Comment by Vadims06** [comment]
> > Hi @rp-nz,
> > the syntax should be like this
> > ```
> > 
> >   flask:
> >     container_name: flask
> >     image: vadims06/topolograph:2.14
> >     restart: unless-stopped
> >     environment:```
> > ```
> 
> Thanks that did the trick. Trying out SROS OSPF DB now. Seems to be working pretty good.

I'm glad to hear this. I'm going to implement Napalm support for this driver soon. So I would be grateful if someone tries and confirms that it works. I notify about adding Nokia's NAPALM support later

**Comment by Vadims06** [comment]
> Are there plans to support IS-IS DB to this tool and also any way we can add node names to each of the nodes in the DB.
> 
> Thanks

No, ISIS support is not planned.
Sure, you can change node names in DNS hostnames Tab, or set DNS IP address in docker-compose.override.yml

**Comment by Vadims06** [comment]
Hi @laimaretto, @hellt,
I added NAPALM support for SROS into topolograph:2.14. It means that it's possible to  set```NAPALM_USERNAME: cisco
        NAPALM_PASSWORD: cisco``` in docker-compose.override.yml and then add router IP in stead of uploading OSPF LSDB in txt file. After this when you choose the router and press Load Hosts - NAPALM driver fetch LSA1, LSA2, LSA5 by itself.
If it doesn't work properly - please send me docker log flask output to the email. Thanks.

**Comment by hellt** [comment]
Cool, thanks!

**Comment by Vadims06** [comment]
Hi @hellt, @laimaretto , I'm just wondering - did you have a chance to try a fetch the network state via NAPALM driver from topologtaph? Probably you faced with some issues or something else?,,
Thanks.

**Comment by Vadims06** [comment]
Nokia support was included in [v2.14](https://github.com/Vadims06/topolograph/tree/v2.14). If you faced with any issues - please do not hesitate to open it again.

**Comment by Vadims06** [comment]
Hi @hellt @laimaretto @rp-nz 
I would like to share with you and offer another extension of Topolograph - [OSPF Watcher](https://github.com/Vadims06/ospfwatcher). The watcher connects to a real network via GRE tunnel and logs any changes in the topology. Logs then are exported to Elastic Search stack where can be filtered by the required time period. So you can always find out what has been happening with your topology in the past.
